### PR TITLE
Handle overflow of test bucketName and reduce dependency on username

### PR DESCRIPTION
**Pull request by vinayakumarb**
https://github.com/GoogleCloudDataproc/hadoop-connectors/blob/master/gcsio/src/test/java/com/google/cloud/hadoop/gcsio/integration/GoogleCloudStorageTestHelper.java#L257

`makeBucketName()` considers `username` as well while constructing unique prefix for bucketname. While its limited to only  9 chars, there is no check to limit overall bucketname to 63 chars.

This PR adds a check and following changes to keep same behavior irrespective of usernames' length in bucket names.

1. Generate the unique string of length 13 chars for the bucket consisting of 
     1. Use first 8 characters of username in bucketname. If username is shorter, use complete username. i.e. No padding.
     2. Out of 12 char, after username (upto 8 chars) remaining are chosen from UUID. (min 4 chars)
4.  `username` and `UUID` together makes upto 15 characters including delimiters.
5. Remaining 48 characters can be utilized by `bucketPrefix` and `suffix` provided by the tests. So keep the cumulative length of `bucketPrefix` and `suffix` provided in `getUniqueBucketName(suffix)` to 48 characters. 

Reason to restrict cumulative instead of individual prefix and suffix : There are existing tests, where suffix of upto 34 chars is used in which provided prefix is only 10 chars. In other tests provided prefix is of bigger size and suffix is small.

In this way, behavior of test results doesnot depend on length of usernames.

### CODE REVIEW

**Comment by vinayakumarb** [comment]
> @vinayakumarb Could you please fix the check style violations ?
> 
> ```
> Run 'mvn spotless:apply' to fix these violations.
> ```

Have applied this and pushed changes again.
 Please review.

**Comment by Deependra-Patel** [comment]
@vinayakumarb Tests are failing with error . PTAL.

"[ERROR] com.google.cloud.hadoop.gcsio.GoogleCloudStorageFileSystemIntegrationTest  Time elapsed: 1.114 s  <<< ERROR!
java.lang.IllegalStateException: uniqueBucketPrefix should start with bucketPrefix"

File: gcsio/src/test/java/com/google/cloud/hadoop/gcsio/integration/GoogleCloudStorageTestHelper.java [hunk 1]
```diff
@@ -256,12 +264,23 @@ public TestBucketHelper(String bucketPrefix) {
...
     private static String makeBucketName(String prefix) {
       String username = System.getProperty("user.name", "unknown").replaceAll("[-.]", "");
-      username = username.substring(0, min(username.length(), 9));
+      // Only first 6 characters of username is considered.
+      username = username.substring(0, min(username.length(), 6));
       String uuidSuffix = UUID.randomUUID().toString().substring(0, 6);
-      return prefix + DELIMITER + username + DELIMITER + uuidSuffix;
+      // [6 char username] + DELIMITER + [6 char UUID] => 13 chars of unique Id
+      // Leaving room of 48 characters for bucketPrefix and suffix (excluding delimiters)
+      return prefix + username + DELIMITER + uuidSuffix;
```
Comment (Deependra-Patel) [inline gcsio/src/test/java/com/google/cloud/hadoop/gcsio/integration/GoogleCloudStorageTestHelper.java:272]: Let's keep in same order

File: gcsio/src/test/java/com/google/cloud/hadoop/gcsio/integration/GoogleCloudStorageTestHelper.java [hunk 2]
```diff
@@ -256,12 +264,23 @@ public TestBucketHelper(String bucketPrefix) {
 
     private static String makeBucketName(String prefix) {
       String username = System.getProperty("user.name", "unknown").replaceAll("[-.]", "");
-      username = username.substring(0, min(username.length(), 9));
+      // Only first 6 characters of username is considered.
+      username = username.substring(0, min(username.length(), 6));
```
Comment (Deependra-Patel) [inline gcsio/src/test/java/com/google/cloud/hadoop/gcsio/integration/GoogleCloudStorageTestHelper.java:268]: Can we keep here 9 (becuase if we take only 6 then there could be confusion like deepan(for deepanshu) and deepen(for deependra)) as before and change the 48 check below to lower number?

File: gcsio/src/test/java/com/google/cloud/hadoop/gcsio/integration/GoogleCloudStorageTestHelper.java (hunk 1 above)
Comment (vinayakumarb) [reply gcsio/src/test/java/com/google/cloud/hadoop/gcsio/integration/GoogleCloudStorageTestHelper.java:272]: It is in same order. Its just that `prefix` already have DELIMITER at the end. So duplication is removed.

File: gcsio/src/test/java/com/google/cloud/hadoop/gcsio/integration/GoogleCloudStorageTestHelper.java (hunk 2 above)
Comment (vinayakumarb) [reply gcsio/src/test/java/com/google/cloud/hadoop/gcsio/integration/GoogleCloudStorageTestHelper.java]: Considering upto 8 chars of username and remaining are being filled with unique UUID.
There may be cases all 8 characters are same for 2 different users, But UUID ensures prefix is unique.

**Comment by vinayakumarb** [comment]
> 

I think this is fixed now.

File: gcsio/src/test/java/com/google/cloud/hadoop/gcsio/integration/GoogleCloudStorageTestHelper.java [hunk 3]
```diff
@@ -246,22 +246,44 @@ public static class TestBucketHelper {
...
       return prefix + DELIMITER + username + DELIMITER + uuidSuffix;
     }
 
+    /**
+     * @param suffix Suffix to be added to the bucket name. Max cumulative length of bucketPrefix
+     *     and suffix is 48 chars.
+     * @return An unique bucket name with specified prefix and suffix.
+     */
     public String getUniqueBucketName(String suffix) {
+      checkArgument(
```
Comment (Deependra-Patel) [inline gcsio/src/test/java/com/google/cloud/hadoop/gcsio/integration/GoogleCloudStorageTestHelper.java:284]: Checking total length of `uniqueBucketPrefix + DELIMETER + suffix` < 63 would have also worked.

**Review by Deependra-Patel — APPROVED** [review]

File: gcsio/src/test/java/com/google/cloud/hadoop/gcsio/integration/GoogleCloudStorageTestHelper.java (hunk 3 above)
Comment (vinayakumarb) [reply gcsio/src/test/java/com/google/cloud/hadoop/gcsio/integration/GoogleCloudStorageTestHelper.java:284]: I agree that this check serves the purpose. But again actual size of `uniqueBucketPrefix` depends on username of running env. Not fixed.
So checking on `bucketPrefix` and `suffix` which are controlled in code is better.

**Pull request merged**